### ISSUE TITLE
Store sent messages

### ISSUE THREAD

**Issue by dbrgn** [saltyrtc/saltyrtc-client-js#62]
...so that when logging a send-error message, we actually know which message was lost.

**Comment by lgrahl** [comment]
I'd only do that in debug mode though.

**Comment by dbrgn** [comment]
@lgrahl the Java version implements that currently:

- https://github.com/saltyrtc/saltyrtc-client-java/blob/master/src/main/java/org/saltyrtc/client/helpers/MessageHistory.java
- https://github.com/saltyrtc/saltyrtc-client-java/blob/034b7daee090bd115f8f400ffb274c050ebecd91/src/main/java/org/saltyrtc/client/signaling/Signaling.java#L1147-L1153

Do you think it should be removed?

**Comment by lgrahl** [comment]
I think it should be active in debug mode only.

**Comment by dbrgn** [comment]
Reason? Performance?

**Comment by lgrahl** [comment]
No, I just want to be cautious when giving out detailed error reasons. Think of the padding oracle attack on SSL.

**Comment by lgrahl** [comment]
I misinterpreted this thread's topic (dunno why). Sure, logging these messages should be fine.

What I was referring to were WebSocket close code reasons.
